src/utils/mod.rs

```diff
@@ -107,9 +108,11 @@ where
     F: FnOnce() -> Result<R> + Send + 'static,
     R: Send + 'static,
 {
-    tokio::task::spawn_blocking(f)
-        .await
-        .context("failed to join thread")?
+    match tokio::task::spawn_blocking(f).await {
+        Ok(result) => result,
+        Err(err) if err.is_panic() => panic::resume_unwind(err.into_panic()),
```

Comment:
Ah yeah, I forgot tokio didn't do this by default, too used to `futures`' join-handles which do.